command/server/config.go

```diff
@@ -97,6 +97,8 @@ type Config struct {
 	LogRequestsLevel    string      `hcl:"-"`
 	LogRequestsLevelRaw interface{} `hcl:"log_requests_level"`
 
+	DetectDeadlocks string `hcl:"detect_deadlocks"`
```

Comment:
I wonder if we should just make this a bool? Right now there's only one valid option, `statelock`, but I don't know of any additional values we may add in the future. Thoughts @ncabatoff?

Reply:
I could see us adding `expiration` for the locks in ExpirationManager.  I think that's the main other lock we've had issues with, but there are plenty of other areas where at some point we might start having problems and decide we want to expand scope for, e.g. replication FSM, quotas, etc.